### ISSUE TITLE
vmagent doesn't collect metrics after redirect with 301

### ISSUE THREAD

**Issue by YuriGrigorov** [VictoriaMetrics/VictoriaMetrics#869]
**Describe the bug**
Starting from version 1.41.1 vmagent doesn't collect metrics from targets which responds with 301.

**To Reproduce**
Save python code to file ```/tmp/redirect.py```
```
#!/usr/bin/python3

from http.server import HTTPServer, BaseHTTPRequestHandler


class Redirect(BaseHTTPRequestHandler):

    def do_GET(self):
        if self.path == '/metrics/':
            self.send_response(200)
            self.send_header('content-type', 'text/plain')
            self.end_headers()
            self.wfile.write(b"vmagent_test_metric 1.0")
        else:
            self.send_response(301)
            self.send_header('content-type', 'text/html')
            self.send_header('Location', '/metrics/')
            self.end_headers()


def run(server_class=HTTPServer, handler_class=Redirect):
    server_address = ('', 8000)
    httpd = server_class(server_address, handler_class)
    httpd.serve_forever()


run()
```

Start web server
```
systemd-run /usr/bin/python3 /tmp/redirect.py
```

Start vmagent **v1.41.0** with config
```
global:
  scrape_interval: 15s
  evaluation_interval: 15s

  - job_name: 'vmagent-test-metric'
    metrics_path: /metrics
    static_configs:
      - targets:
        - localhost:8000
```

Get values from storage
```
curl -s -g 'http://vmselect:8481/select/1/prometheus/api/v1/query_range?query=vmagent_test_metric&start=-1m&step=15'

{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "vmagent_test_metric",
          "instance": "localhost:8000",
          "job": "vmagent-test-metric"
        },
        "values": [
          [
            1603916508.454,
            "1"
          ],
          [
            1603916523.454,
            "1"
          ],
          [
            1603916538.454,
            "1"
          ],
          [
            1603916553.454,
            "1"
          ]
        ]
      }
    ]
  }
}
```

Upgrade any to version after v1.41.0 (ex. 1.44.0)

Start vmagent

Get values from storage
```
curl -s -g 'http://vmselect:8481/select/1/prometheus/api/v1/query_range?query=vmagent_test_metric&start=-1m&step=15'

{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": []
  }
}
```

**Expected result**
4 items in values array

**Comment by valyala** [comment]
@YuriGrigorov , the bug should be fixed in the commit c0f3be824d32aed3fa4bdabf4820f1cf3508abc1 .

**Comment by valyala** [comment]
The bugfix has been included in [v1.45.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). @YuriGrigorov , could you confirm that `vmagent` works as expected for your case?

**Comment by YuriGrigorov** [comment]
Thank you! After upgrading ```vmagent``` to v1.45.0 it works as expected.
